rules/endpoints.go

```diff
@@ -220,6 +249,19 @@ func (r *DefaultRuleRenderer) endpointIptablesChain(
 		}
 	}
 
+	if chainType == chainTypeSetEndPointMark {
```

Comment:
I think it'd be better to move this to a new function, only reason we share the code below is that the chain has common structure but this type of chain is very different.

Reply:
Agree.